### PR TITLE
Federation

### PR DESCRIPTION
**Pull request by wojtekLs**

### CODE REVIEW

File: config/federationConfigClass.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	openApiEndServiceWorkerCA() {
+		const SwRegistration = async () => {
+			if (worker in navigator) {
+				try {
+					const reg = await navigator.serviceWorker.register(worker);
+				} catch (e) {
+					winston.error(workerRegistrationError);
+				}
+			} else {
+				winston.error(noServiceWorkerExist);
```
Comment (wojtekxtx) [inline config/federationConfigClass.js:31]: See [#4](https://github.com/wojtekxtx/ejs/issues/4)

File: config/federationConfigClass.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		// @todo: #5 add metadata proccessing code
+	}
+
+	/**
+	 * Writes schema to file
+	 * 
+	 * @param {} federationSchemaName
+	 * @param {*} federationSchemaFile
+	 * 
+	 * @todo #6 Upload schema file to SchemaStore
```
Comment (wojtekLs) [inline config/federationConfigClass.js:96]: We may need to discuss it further.
```suggestion
	 * @todo #6 Upload schema file to SchemaStore
```

File: config/federationConfigClass.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+			success: (data) => {
+				let binary = Buffer.from(data, "base64");
+				let array = [];
+				for (let i = 0; i < binary.length; i++) {
+					array.push(binary[i]);
+				}
+				federationSchemaFile = new Blob([new Uint8Array(array)], { type: 'application/json' });
+			},
+			fail: () => {
+				winston.error(errorSettingSchema);
```
Comment (wojtekLs) [inline config/federationConfigClass.js:113]:
```suggestion
				winston.error(errorSettingSchema);
```
```suggestion
				winston.error(errorSettingSchema);
```edit it to log to `stdout` instead

**Review by wojtekLs — COMMENTED** [review]
LGTM :)

**Pull request merged**